main.py

```diff
@@ -78,7 +78,25 @@ def setCheckAttr(widget, var): self.settings.settings[var] = widget.isChecked()
 		def getCheckAttr(widget, var): widget.setChecked(self.settings.settings[var])
 		def setDropdownAttr(widget, var): self.settings.settings[var] = widget.currentText()
 		def getDropdownAttr(widget, var): widget.setCurrentText(self.settings.settings[var])
-		def updateLaunchCmd(): self.invocationPreview.setPlainText(Xqemu.launchCmdToString(Xqemu.generateLaunchCmd(self.settings, True)))
+
+		def updateLaunchCmd():
+			controller_arr = [
+				['controller_one', self.controller1],
+				['controller_two', self.controller2],
+				['controller_three', self.controller3],
+				['controller_four', self.controller4]
+			]
+			for i, controller in enumerate(controller_arr, 1):
+				for ab in ['A', 'B']:
+					if settings.settings[controller[0]] == 'Not connected':
```

Comment:
Instead of checking for "Not connected", we should probably look for some index / key?

This looks scary when considering translations.

*Not sure if this is a problem with this PR, or a general issue in our codebase. If it's an issue in the codebase, we should have an issue for it.*